src/kagglehub/clients.py

```diff
@@ -43,7 +44,28 @@
   {}
 """
 
-KAGGLEHUB_USER_AGENT = {"User-Agent": f"kagglehub/{kagglehub.__version__}"}
+_cached_user_agent = None
+
+
+def get_user_agent() -> str:
+    global _cached_user_agent  # noqa: PLW0603
+    if _cached_user_agent:
+        return _cached_user_agent
+
+    base_user_agent = f"kagglehub/{kagglehub.__version__}"
+
+    if is_in_kaggle_notebook():
+        build_date = read_kaggle_build_date()
+        _cached_user_agent = f"{base_user_agent} kkb/{build_date}"
+    elif is_in_colab_notebook():
+        colab_tag = os.getenv("COLAB_RELEASE_TAG")
```

Comment:
@mayankmalik-colab Can you review this user agent for Colab to make sure we got this right. TY